include/zephyr/drivers/battery.h

```diff
@@ -0,0 +1,470 @@
+
+/**
+ * Calculate battery time in minutes, under an assumed current.
+ *
+ * @param rate		Current to use for calculation, in mA.
+ *			If > 0, calculates charging time; if < 0, calculates
+ *			discharging time; 0 is invalid and sets minutes=0.
+ * @param minutes	Destination for calculated time in minutes.
+ * @return non-zero if error.
+ */
+typedef int (*battery_time_at_rate)(const struct device *dev, int rate, int *minutes);
+
+/**
+ * Read battery charge cycle count.
+ *
+ * @param count		Destination for count.
+ * @return non-zero if error.
+ */
+typedef int (*battery_cycle_count)(const struct device *dev, int *count);
+
+/**
+ * Read battery manufacture date.
+ *
+ * @param year		Destination for year
+ * @param month		Destination for month
+ * @param day		Destination for day
+ * @return non-zero if error.
+ */
+typedef int (*battery_manufacture_date)(const struct device *dev, int *year, int *month, int *day);
+
+/**
+ * Read battery serial number.
+ *
+ * @param serial	Destination for serial number.
+ * @return non-zero if error.
+ */
+typedef int (*battery_serial_number)(const struct device *dev, int *serial);
+
+/**
+ * Read manufacturer name.
+ *
+ * @param dest		Destination buffer.
+ * @param size		Length of destination buffer in chars.
+ * @return non-zero if error.
+ */
+typedef int (*battery_manufacturer_name)(const struct device *dev, char *dest, int size);
+
+/**
+ * Read manufacturer name.
+ *
+ * This can be overridden to return a chip or board custom string.
+ *
+ * @param dest		Destination buffer.
+ * @param size		Length of destination buffer in chars.
+ * @return non-zero if error.
+ */
+typedef int (*get_battery_manufacturer_name)(const struct device *dev, char *dest, int size);
+
+/**
+ * Read device name.
+ *
+ * @param dest		Destination buffer.
+ * @param size		Length of destination buffer in chars.
+ * @return non-zero if error.
+ */
+typedef int (*battery_device_name)(const struct device *dev, char *dest, int size);
+
+/**
+ * Read battery type/chemistry.
+ *
+ * @param dest		Destination buffer.
+ * @param size		Length of destination buffer in chars.
+ * @return non-zero if error.
+ */
+typedef int (*battery_device_chemistry)(const struct device *dev, char *dest, int size);
+
+/**
+ * Read device manufacture date.
+ *
+ * @param year		Destination for year
+ * @param month		Destination for month
+ * @param day		Destination for day
+ * @return non-zero if error.
+ */
+typedef int (*battery_manufacturer_date)(const struct device *dev, int *year, int *month, int *day);
+
+/**
+ * Report the absolute difference between the highest and lowest cell voltage in
+ * the battery pack, in millivolts.  On error or unimplemented, returns '0'.
+ */
+typedef int (*battery_imbalance_mv)(const struct device *dev);
+
+/**
+ * Call board-specific cut-off function.
+ *
+ * @return EC_RES_INVALID_COMMAND if the battery doesn't support.
+ */
+typedef int (*board_cut_off_battery)(const struct device *dev);
```

Comment:
suggest name `battery_cut_off`

Reply:
... or someone is getting the scissors out :-)